### ISSUE TITLE
Add TimeStamp to PlaceMark?

### ISSUE THREAD

**Issue by blipmusic** [georust/kml#23]
EDIT: I completely missed to mention that this concerns serialization, apologies!

Are there plans to support timestamps for `Placemark` in the future (and perhaps elsewhere where valid?).

If e.g.
```
<TimeStamp>
    <when>2022-04-02T10:00:00</when>
</TimeStamp>
```
is added as a custom `Element` it will be incorrectly positioned, meaning validation against the KML v2.2 schema will fail. I think it is expected to follow immediately after `<description>`.

I had similar positional issues when trying out CDATA as a custom `<description>` `Element`, but ended up removing the CDATA tag and just set `Placemark.description` to the html I wanted, since quick-xml escapes structural characters in "value positions" anyway.

**Comment by pjsier** [comment]
@blipmusic thanks for this issue as well! Ideally I'd like to support all valid `kml` types but have mostly prioritized geometries.

I'm still going through the specification for this part, but would all elements including `TimeStamp` and others go after `description` and before `coordinates`? If so, we could move the writing of custom elements into that spot and users could be responsible for custom `Element` objects being the correct order. If not I'm open to and PRs for adding a `TimeStamp` type!

**Comment by pjsier** [comment]
Thanks for the additional info! I think for now it would be best to at least not force the user to generate KML that would be invalid against the spec, so would updating `write_placemark` to something like this work until the additional elements are created?

```rust
    fn write_placemark(&mut self, placemark: &Placemark<T>) -> Result<(), Error> {
        self.writer
            .write_event(Event::Start(BytesStart::owned_name(b"Placemark".to_vec())))?;
        if let Some(name) = &placemark.name {
            self.write_text_element(b"name", name)?;
        }
        if let Some(description) = &placemark.description {
            self.write_text_element(b"description", description)?;
        }
        for c in placemark.children.iter() {
            self.write_element(c)?;
        }
        if let Some(geometry) = &placemark.geometry {
            self.write_geometry(geometry)?;
        }
        Ok(self
            .writer
            .write_event(Event::End(BytesEnd::borrowed(b"Placemark")))?)
    }
```

If we know geometry needs to be last and everything else is after description then users can handle the order of any elements that aren't officially supported with `Element`. Would that work?

**Comment by pjsier** [comment]
I'm considering this initial issue closed by #25 since at least now we aren't preventing users from generating correct KML, but I think a separate issue for adding these elements with full support could make sense. Thanks again!
